si/session.py

```diff
@@ -213,6 +211,7 @@ def multi_decide_with_context(self, context_name: str, properties, decisions, **
                 reply["decisions"][i] = decision
             elif "decision" in decision and decision["decision"] != '':
                 reply["decisions"][i] = {"decision": json.loads(decision["decision"]),
+                                         "intent": decision.get("intent"),
```

Comment:
This will return `None` when `intent` is not part of the decision dict. This may be different from the original approach.